include/deal.II/base/numbers.h

```diff
@@ -318,7 +318,7 @@ namespace numbers
   typename NumberTraits<number>::real_type
   NumberTraits<number>::abs (const number &x)
   {
-    return std::fabs(x);
+    return std::fabs((double)x);
```

Comment:
Yes, this doesn't work. How about just calling `std::abs()`? This should be overloaded for all floating point types.